tools/check.sh

```diff
@@ -1,5 +1,11 @@
 #!/bin/bash
 
+# Filter that exits the process if its input is non-empty.
+# If $* args given, prefixes each line of input with "Error: $*: ".
+fail() {
+  awk -v str="$*" '/./ { if (length(str)) { print "Error: " str ": " $0 } x=1 } EXIT { exit x }'
```

Comment:
This doesn't exit the surrounding shell, it just exits the awk script. Thus, this changes the status quo. But maybe we want all checks run, and not terminate after the first error.

But the general idea is nice. We should mix it with the "github action log commands" so that the errors are categorized as such, and colored red. Give me a little while.

Reply:
Oops, I lost the `|| exit 1` when changing it from a sed command (as demo'd at https://github.com/cplusplus/draft/issues/4529#issuecomment-786559241) to an awk one.

Reply:
Oops, but it needs to be `END` not `EXIT`

Reply:
And it still won't work, because it exits the process at the end of the pipeline, which doesn't exit the process running the script. We need `set -e` to make it exit the whole script.